lib/common_fw/common_fw.cpp

```diff
@@ -9,24 +9,28 @@ SdFat sd;
 File file;
 struct Queue *sdBuffer;
 union floatArrToBytes farrbconvert;
+int packetCounter = 0;
 
 /**
  *
  */
 bool write_to_SD(std::string message, const char * file_name) {
+    using namespace std;
     sdBuffer->enqueue(message);
     if(sdBuffer->length >= 40) {
-      if(file.open(file_name, O_RDWR | O_APPEND)) {
+      fstream currfile;
+      currfile.open(file_name, fstream::out | fstream::app);
```

Comment:
Not sure I understand why you did this. We have `file` defined in config

Reply:
apparently file.write() doesn't exist in c++ - have to use fsteam to write to files (so I opened the file w fstream to make sure it wouldn't error)

Reply:
or is this an arduino thing

Reply:
It's an arduino thing. it should work if you're using pio to compile